poetry/core/masonry/api.py

```diff
@@ -52,14 +52,14 @@ def prepare_metadata_for_build_wheel(metadata_directory, config_settings=None):
 
 def build_wheel(wheel_directory, config_settings=None, metadata_directory=None):
     """Builds a wheel, places it in wheel_directory"""
-    poetry = Factory().create_poetry(Path(".").resolve())
+    poetry = Factory().create_poetry(Path(".").resolve(), with_dev=False)
```

Comment:
The verification of this shouldbe done by duplcating the existing test with the new fixtures added here. File content verification is not required, but just need to verify that a) wheel buidls when invalid dev dependency encountered and b) wheel builds will fail if these are encountered in the main category.

https://github.com/python-poetry/poetry-core/blob/0db07bc81b989c5c4933cf019fec7a4f93fbd744/tests/masonry/test_api.py#L46-L54

It might also be great to add an integration test case if easy enough. See the integration test suite at https://github.com/python-poetry/poetry-core/blob/master/tests/integration/test_pep517.py.

Reply:
I think for edge cases not meant to affect the build otherwise, it should stay in the lower level and not in the integration test. Let me know if you feel strongly about that though.

Reply:
This is still a case, not an edge case per say. We are verifying that packages with invalid or non-existent dev dependnecies can still be built. This is not an edge case. 

Integration testing (`pip` or `pep517` call) is option for this, but api (masonry) level testing is mandatory.